### DISCUSSION TITLE
Open results in new tab & Select multiple results for download

### DISCUSSION THREAD

**Discussion by SPLeon** [microsoft-search/pnp-modern-search#2932] [Ideas]
Hi,

I just started to use the PnP Search addon. Very impressive addon :-)

I have a couple of questions regarding the functionality:

1. Is it possible to open an item within the search result webpart in a new tab? I'm using the detail list but I dont see an option to open a specific result in a new tab.

2. Our company has asked if it's possible to select mutiple items in the result webpart and download the selected items. Is this also possible? We don't write custom code as we are not experience in this so we prefer a OOTB option.

Many thanks.

**Comment by FranckyC** [comment]
Hi @SPLeon:

> Is it possible to open an item within the search result webpart in a new tab? I'm using the detail list but I dont see an option to open a specific result in a new tab.

Yes by modifying the "Title" column HTML value and using the `data-interception="off"` attribute:

![image](https://user-images.githubusercontent.com/5659122/110476231-bdfc6900-80af-11eb-8f77-8844745c95b0.png)

> Our company has asked if it's possible to select mutiple items in the result webpart and download the selected items. Is this also possible? We don't write custom code as we are not experience in this so we prefer a OOTB option.

Not available for any layout for now. Maybe we will implement this in a next release but nothing planned yet.

Thanks!

**Comment by SPLeon** [comment]
Thanks for the quick reply. Opening in new tab is working now.
Is there also a possibility to get a preview of a file within this detail view?

**Comment by FranckyC** [comment]
Yes, but you have to update the column value and use the custom component `<pnp-filepreview>`. See https://microsoft-search.github.io/pnp-modern-search/extensibility/web_components_list/#ltpnp-filepreviewgt. Refer to the simple list layout for usage example.

**Comment by fedewagner** [comment]
Hi, 

is there any news about the plan to include the **download button** for multiple selected files from the Result Web Part in the next release? 

Have you seen any other customization for enabling that?

Any other further information will be appreciated.

**Comment by SylvieLet** [comment]
I'm also super interested to know if there's been any development on the possibility of downloading multiple selected files from the Results web part.

Anyone know of such a customisation?

**Comment by fedewagner** [comment]
As a temporary solution, I included the "Download" feature as a button so that the downloading can be done easily and faster:

![image](https://user-images.githubusercontent.com/64759519/235634047-3aa112c1-bed7-495e-9d18-db0e36db1879.png)

![image](https://user-images.githubusercontent.com/64759519/235634132-53925ab7-6383-411d-a25a-a8f3012ce849.png)

`<a href="{{slot item @root.slots.Path}}" download>
   <input type="submit" formtarget="_blank" style="width:100%" value="🡳 Download file"/>
 </a>`
 
I hope it's useful for you.

**Reply by nthoman** [reply]
Love this Download work around! Thanks for sharing!

**Comment by MaruVC** [comment]
Hi!
Is there a way to create a link to the file location, but getting it to open on a new tab?
I am using this:
"<a href="{{slot item @root.slots.PreviewUrl}}" target="_blank" style="color: {{@root.theme.semanticColors.link}}">{{slot item@root.slots.Title}}</a>"
But it opens the file location on the search page.

The other thing i've been ask to do, is to get the file to open on Desktop app (at the moment it opens them on Browser) 
I checked the site settings and it is to set in desktop by default but it only seems to work if i open the file from the document library not from the Search result web part.
Currently i have this "<a href="{{slot item @root.slots.PreviewUrl}}" target="_blank" style="color: {{@root.theme.semanticColors.link}}">{{slot item @root.slots.Title}}</a>" and the file opens on the browser (same page as the search)
I tried with web=0 but that downloads the file.
If you have any tips, please let me know.
Thank you

**Reply by wobba** [reply]
You need to get target="_blank" into the markup to get a new tab in addition to having `data-interception="off"` as per https://learn.microsoft.com/en-us/sharepoint/dev/spfx/hyperlinking
